pyproject.toml

```diff
@@ -60,6 +45,7 @@ maintainers = [
 #     https://scipy.github.io/devdocs/dev/core-dev/index.html#version-ranges-for-numpy-and-other-dependencies
 requires-python = ">=3.8"
 dependencies = [
+    # TODO: update to "pin-compatible" once possible, see https://github.com/FFY00/meson-python/issues/29
```

Comment:
No, it should be a string that `meson-python` dynamically replaces to `>=<version-at-build_time>`. So if you build against numpy 1.18.5, this will say `>=1.18.5` (now our lowest-supported version), if you build against 1.22.1, then `>=1.22.1`, etc. This will solve a long-standing issue where our `install_requires` for `numpy` is simply wrong when users build wheels, or even when we build wheels for PyPI where due to Python or platform version support the numpy version we build against is not the lowest-supported one.

The actual syntax of this still needs to be finalized, see https://github.com/FFY00/mesonpy/issues/29.

Reply:
That sounds a bit like what you'd get from using setup_requires with unversioned numpy, and then an install_requires that is dynamically calculated using homebrew functions in setup.py based on whichever version got installed... which setuptools may not have ever supported because you'd need to calculate that after the build requirements got installed and calculated. At least it would definitely work if you installed numpy before running ye olde `python3 setup.py install`.

I can see why you'd want proper support for that in the build backend, lol.